src/store/storage-controller.ts

```diff
@@ -111,19 +109,53 @@ export class StorageController {
     return messages;
   }
 
-  public static async delete(
-    messageStore: MessageStore,
-    dataStore: DataStore,
+  public async delete(
     tenant: string,
     message: BaseMessage
   ): Promise<void> {
     const messageCid = await Message.getCid(message);
 
     if (message.descriptor.dataCid !== undefined) {
-      await dataStore.delete(tenant, messageCid, message.descriptor.dataCid);
+      await this.dataStore.delete(tenant, messageCid, message.descriptor.dataCid);
     }
 
-    await messageStore.delete(tenant, messageCid);
+    await this.messageStore.delete(tenant, messageCid);
+  }
+
+  public getMessage(tenant: string, messageCid: string): Promise<BaseMessage|undefined> {
+    return this.messageStore.get(tenant, messageCid);
+  }
+
+  public putMessage(
+    tenant: string,
+    messageJson: BaseMessage,
+    indexes: Record<string, string>,
+    options?: MessageStoreOptions): Promise<void> {
+    return this.messageStore.put(tenant, messageJson, indexes, options);
+  }
+
+  public queryMessages(tenant: string, filter: Filter, options?: MessageStoreOptions): Promise<BaseMessage[]> {
+    return this.messageStore.query(tenant, filter, options);
+  }
+
+  public getData(tenant: string, messageCid: string, dataCid: string): Promise<GetResult|undefined> {
+    return this.dataStore.get(tenant, messageCid, dataCid);
+  }
+
+  public appendEvent(tenant: string, messageCid: string): Promise<string>|undefined {
+    return this.eventLog.append(tenant, messageCid);
+  }
+
+  public deleteEvents(tenant: string, deletedMessageCids: any): Promise<number>|undefined {
```

Comment:
should this be `Array<string>`?

also, i wonder if TypeScript has a way to do a "passthrough" for scenarios like this where we dont have to declare the types 🤔

Reply:
Addressed in https://github.com/TBD54566975/dwn-sdk-js/pull/338/commits/931ab6bd4e5079229f6b9663de5a442eedfe4983